### PR TITLE
update shellcheck

### PR DESCRIPTION
**Pull request by charlierudolph**
@allewun @kevgo

### CODE REVIEW

File: bin/lint [hunk 1]
```diff
@@ -1,6 +1,12 @@
 #!/usr/bin/env bash
 set -e
 
-find src -name 'git-*' -print0 | xargs -0 shellcheck --exclude=SC2155
-find src -name '*.sh' -print0 | xargs -0 shellcheck --exclude=SC2155
-find bin -type f -print0 | xargs -0 shellcheck --exclude=SC2155
+# Exluded rules (see https://github.com/koalaman/shellcheck/wiki/)
```
Comment (allewun) [inline bin/lint]: Ex[c]luded

File: bin/lint [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+#
+# SC1090: we aready run shellcheck on all our source files
```
Comment (allewun) [inline bin/lint]: a[l]ready

File: bin/lint [hunk 3]
```diff
@@ -1,6 +1,12 @@
...
+# SC2155: would bloat the code base
+# SC2162: read -r option not available on Mac OSX 10.10
+
+exclude="SC1090,SC2155,SC2162"
+
+find bin src -type f ! -name 'README.md' -print0 | xargs -0 shellcheck --exclude="$exclude"
```
Comment (allewun) [inline bin/lint]: `find bin src -type f ! -iregex '.*\.md' -print0 | ...` to be a tiny bit more robust

**Comment by allewun** [comment]
:ship:

**Comment by kevgo** [comment]
Very cool to see ShellCheck being so actively developed, and this change adds better infrastructure to handle it! :ship:

**Pull request closed without merging**